src/cluster_legacy.c

```diff
@@ -3883,6 +4044,42 @@ clusterMsgSendBlock *clusterCreatePublishMsgBlock(robj *channel, robj *message,
     return msgblock;
 }
 
+clusterMsgSendBlock *clusterCreatePublishLightMsgBlock(robj *channel, robj **messages, int count, uint16_t type) {
+    uint32_t channel_len, message_len;
+    int i;
+
+    channel = getDecodedObject(channel);
+    channel_len = sdslen(channel->ptr);
+
+    uint32_t aggregated_msg_len = 0;
+    aggregated_msg_len += channel_len;
+    for (i = 0; i < count; i++) {
+        messages[i] = getDecodedObject(messages[i]);
+        message_len = sdslen(messages[i]->ptr);
+        aggregated_msg_len += message_len;
+    }
+
+    size_t msglen = sizeof(clusterMsgLight) - sizeof(union clusterMsgDataLight);
+    msglen += sizeof(clusterMsgDataPublishMulti);
```

Comment:
why do we include the entire struct `clusterMsgDataPublishMulti`? `bulk_data` is an in-memory pointer. The pointer size varies on 32 platforms vs 64 platforms. I don't think people would run mixed 32bit and 64bit server builds in the same cluster. but it is technically incorrect to use any platform specific data types. Nevertheless, I think we can save some memory too.

```suggestion
    msglen += offsetof(clusterMsgDataPublishMulti, bulk_data);
```

Reply:
We have removed the Multi publish struct for this PR